packages/core/src/sso/SamlConnector/index.ts

```diff
@@ -181,16 +153,102 @@ class SamlConnector {
    *
    * @returns The raw SAML metadata in XML-format.
    */
-  private async getIdpXmlMetadata() {
-    if (this._rawSamlMetadata) {
-      return this._rawSamlMetadata;
+  private async getIdpMetadataXml() {
+    if (this._samlIdpMetadataXml) {
+      return this._samlIdpMetadataXml;
     }
 
-    const rawSamlMetadata = await getRawSamlMetadata(this.config);
-    if (rawSamlMetadata) {
-      this._rawSamlMetadata = rawSamlMetadata;
+    const samlMetadataXml = await getSamlMetadataXml(this.config);
+    if (samlMetadataXml) {
+      this._samlIdpMetadataXml = samlMetadataXml;
     }
-    return this._rawSamlMetadata;
+
+    return this._samlIdpMetadataXml;
+  }
+
+  /**
+   * Get the manually filled SAML IdP metadata from the raw SAML SSO connector config (including `signInEndpoint` and `x509Certificate`).
+   *
+   * @returns Manually filled SAML IdP metadata.
+   */
+  private async getIdpMetadataJson() {
+    if (this._samlIdpMetadataJson) {
+      return this._samlIdpMetadataJson;
+    }
+
+    // Required fields of metadata should not be undefined.
+    const result = samlIdentityProviderMetadataGuard.safeParse(this.config);
```

Comment:
Should we throw error here? Otherwise the zod guard error context will be lost. Can not which settings a missing.

Reply:
Good call, the previous implementation throws in another place with general info of invalid config. Throw error with a detailed error message right away could be better. Updated.

Reply:
Also need to make sure this exception won't break the `GET /sso-connectors/:connectorId` management API.

Reply:
Confirmed. GET APIs (both GET all and GET by id) catch the error of `getConfig()` method and return `undefined`.